src/payment/coupon/CouponForm.jsx

```diff
@@ -33,16 +33,69 @@ export class CouponForm extends Component {
     this.props.removeCoupon(this.props.voucherId);
   }
 
+  renderInvalidMessage() {
```

Comment:
There may be a more concise way to do this; I didn't want to presume they'd all need the code (even though all the current ones do), otherwise I could have collapsed this down a lot.

Reply:
Just letting it breathe until the code speaks to me and tells me what shape it wants to be. ;)